servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/CompositeCancellable.java

```diff
@@ -59,13 +60,30 @@ private CompositeCancellable(Cancellable... others) {
     public void cancel() {
         if (cancelledUpdater.compareAndSet(this, 0, 1)) {
             if (others == null) {
-                //noinspection ConstantConditions
-                first.cancel();
-                //noinspection ConstantConditions
-                second.cancel();
+                try {
+                    //noinspection ConstantConditions
+                    first.cancel();
+                } finally {
+                    //noinspection ConstantConditions
+                    second.cancel();
+                }
             } else {
```

Comment:
(pre-existing) since there is a static factory method for generation here, consider breaking apart into two separate classes to avoid unused state and duplicate conditionals.